frontend/src/features/rubricBuilder/CSVUpload.tsx

```diff
@@ -1,71 +1,126 @@
+  rubric: Rubric | undefined;
+  setRubric: React.Dispatch<React.SetStateAction<Rubric | undefined>>;
 }
 
-export const CSVUpload: React.FC<CSVUploadProps> = ({
-  onDataChange,
-  closeImportCard,
-}: CSVUploadProps) => {
-  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const file = event.target.files?.[0];
-    if (!file) return;
+export const CSVUpload: React.FC<CSVUploadProps> = ({ setRubric }) => {
+  const [showVersionModal, setShowVersionModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+  const closeErrorDialog = () => setErrorMessage(null);
 
-    if (fileExtension === "csv") {
-      parseCSV(file);
-    } else {
+  const handleFileChange = (
+    file: File,
+    version: "versionOne" | "versionTwo",
+  ) => {
+    const fileExtension = file.name.split(".").pop()?.toLowerCase();
+    if (fileExtension !== "csv") {
       alert("Unsupported file format. Please upload a CSV file.");
+      return;
     }
+
+    importCsv(
+      file,
+      version,
+      (newCriteria) => {
+        updateRubric(newCriteria);
+      },
+      (error) => {
+        setErrorMessage(error);
+      },
+    );
   };
 
-  const parseCSV = (file: File) => {
-    Papa.parse(file, {
-      header: false, // keeps the output an array to sync with parsing xlsx files
-      complete: (results) => {
-        console.log("Parsed CSV data:", results.data);
+  const updateRubric = (newCriteria: Criteria[]) => {
+    setRubric((prevRubric) =>
+      prevRubric
+        ? {
+            ...prevRubric,
+            criteria: [...prevRubric.criteria, ...newCriteria],
+            pointsPossible:
+              prevRubric.pointsPossible +
+              newCriteria.reduce((sum, criterion) => sum + criterion.points, 0),
+          }
+        : {
+            id: parseInt(uuid(), 16),
+            title: "Imported Rubric",
+            criteria: newCriteria,
+            pointsPossible: newCriteria.reduce(
+              (sum, criterion) => sum + criterion.points,
+              0,
+            ),
+            key: `rubric-${uuid()}`,
+          },
+    );
+  };
 
-        // Validate each row to ensure it matches CSVRow type
-        const parsedData = results.data.filter((row): row is CSVRow => {
-          return (
-            Array.isArray(row) &&
-            typeof row[0] === "string" &&
-            row
-              .slice(1)
-              .every(
-                (cell) => typeof cell === "string" || typeof cell === "number",
-              )
-          );
-        });
-        console.log("Validated CSV data:", results.data);
-        onDataChange(parsedData); // Pass validated data to parent
-      },
-    });
-    closeImportCard();
+  const handleVersionSelection = (version: "versionOne" | "versionTwo") => {
+    const fileInput = document.createElement("input");
```

Comment:
we have JSX, no need to manually create DOM elements!

this also could lead to issues with being able to use React features with this stuff

Reply:
example jsx refactor

```jsx
<input
  type="file"
  accept=".csv"
  onChange={(event) => {
    const file = event.target.files?.[0];
    if (file) handleFileChange(file, selectedVersion); // Store `selectedVersion` in state.
  }}
  className="hidden" // Use a visually hidden input if you want a button to trigger it
  ref={fileInputRef} // Use `useRef` to programmatically trigger clicks.
/>
```